### PR TITLE
[IMP] OD12-496, superfluous order line refund invoice line relations

### PR DESCRIPTION
**Pull request by StefanRijnhart**
As the outcome of OD12-496, I'm proposing a different solution to the redundant refund invoice line links that also applies when a recompute is triggered, and not only fixes the situation in the migrated database.

### CODE REVIEW

File: addons/sale/migrations/9.0.1.0/post-migration.py [hunk 1]
```diff
@@ -169,10 +168,52 @@ def set_so_line_computed_rest(env):
...
+    )
+DELETE FROM sale_order_line_invoice_rel
+WHERE invoice_line_id IN (
+    SELECT ail.id FROM account_invoice_line ail
+    JOIN sale_order_line_invoice_rel rel ON rel.invoice_line_id = ail.id
+    JOIN qties ON qties.id = rel.order_line_id
+    JOIN account_invoice ai ON ai.id = ail.invoice_id
+    WHERE ai.type = 'out_refund' AND ai.state != 'cancel'
+    );
+        """)
```
Comment (hbrunn) [inline addons/sale/migrations/9.0.1.0/post-migration.py:209]: do all these invoices have the the good origin field, or how do we know that those invoices still belong to their orders?

Comment (StefanRijnhart) [reply addons/sale/migrations/9.0.1.0/post-migration.py:209]: I checked that all these invoices were linked to the sale order through sale_order_invoice_rel, so yes.

Comment (hbrunn) [reply addons/sale/migrations/9.0.1.0/post-migration.py:209]: but we don't use sale_order_invoice_rel any more, or is the idea that I revert https://github.com/VanMoof/sale-financial/pull/3 to looking at this field too?

Comment (StefanRijnhart) [reply addons/sale/migrations/9.0.1.0/post-migration.py:209]: sale_order_invoice_rel serves as the basis for the invoices' origin updates here: https://github.com/VanMoof/odoo-addons/blob/12.0/scripts/20200326-migrate_all.bash#L133-L136

Comment (hbrunn) [reply addons/sale/migrations/9.0.1.0/post-migration.py:209]: clear

**Review by hbrunn — APPROVED** [review]

**Pull request merged**